apps/spark/src/main/java/com/linkedin/openhouse/jobs/util/TableStatsCollectorUtil.java

```diff
@@ -247,6 +258,43 @@ private static long getSumOfFileSizeBytes(Dataset<Row> allDataFiles) {
         .getLong(0);
   }
 
+  /**
+   * Get the earliest partition date on a table.
+   *
+   * @param table
+   * @param spark
+   * @param partitionColumnName
+   * @return
+   */
+  private static String getEarliestPartitionDate(
+      Table table, SparkSession spark, String partitionColumnName) {
+    if (partitionColumnName == null) {
+      return null;
+    }
+
+    Dataset<Row> partitionData =
+        SparkTableUtil.loadMetadataTable(spark, table, MetadataTableType.PARTITIONS);
+    String partitionColumn = String.format("partition.%s", partitionColumnName);
+
+    String earliestPartitionDate =
+        partitionData
+            .select(partitionColumn)
+            .orderBy(functions.asc(partitionColumn))
+            .first()
+            .get(0)
+            .toString();
+
+    return earliestPartitionDate;
+  }
+
+  private static String getPartitionColumnName(Table table) {
+    return StreamSupport.stream(table.spec().partitionType().fields().spliterator(), false)
+        .filter(field -> field.type() instanceof Types.DateType)
```

Comment:
Is there a better way to get the partition column name? There could be other column with DateType as well.

Reply:
we only allow 1 DateType partition, attempting to partition on 2 `Timestamp` type columns throws this error:
`scala> spark.sql("CREATE TABLE u_selchen.test_mult_partitions(date1 timestamp, date2 timestamp) PARTITIONED BY (days(date1), days(date2))")
java.lang.IllegalArgumentException: OpenHouse only supports 1 timestamp-based column partitioning, 2 were provided: date1_day, date2_day`

Reply:
Yeah make sense. You are already getting the partitionType().fields().